app/routes/_layout+/assets.$assetId.duplicate.tsx

```diff
@@ -0,0 +1,150 @@
+import type { ActionArgs, LoaderArgs } from "@remix-run/node";
+import { json, redirect } from "@remix-run/node";
+import { Form, useLoaderData, useNavigation } from "@remix-run/react";
+import { AssetImage } from "~/components/assets/asset-image";
+import Input from "~/components/forms/input";
+import { Badge, Button } from "~/components/shared";
+import { Spinner } from "~/components/shared/spinner";
+import { db } from "~/database";
+import { duplicateAsset } from "~/modules/asset";
+import { requireAuthSession } from "~/modules/auth";
+import styles from "~/styles/layout/custom-modal.css";
+import {
+  assertIsPost,
+  isFormProcessing,
+  userFriendlyAssetStatus,
+} from "~/utils";
+import { MAX_DUPLICATES_ALLOWED } from "~/utils/constants";
+import { sendNotification } from "~/utils/emitter/send-notification.server";
+import { ShelfStackError } from "~/utils/error";
+
+export const loader = async ({ params }: LoaderArgs) => {
```

Comment:
On every loader or action within the logged in layouts you need to use `requireAuthSession`